### DISCUSSION TITLE
Error setting up Splunk OpenTelemetry Collector for Kubernetes

### DISCUSSION THREAD

**Discussion by oarusoft** [signalfx/splunk-otel-collector-chart#526] [General] [answered]
I am trying to setup OpenTelemetry Collector for Kubernetes in Splunk Cloud.

For this I followed the following article

[Splunk OpenTelemetry Collector for Kubernetes](https://github.com/signalfx/splunk-otel-collector-chart) 

I am using google Kubernetes engine. As mentioned in article I tried both below two helm to  install Splunk OpenTelemetry Collector in a Kubernetes cluster. None of these worked and I keep getting error message " **no such host**".

 

helm install my-splunk-otel-collector --set="splunkPlatform.endpoint=https://http-inputs-prd-p-19lmt.splunkcloud.com:8088/services/collector/event,splunkPlatform.token=XXXX-XXXX-XXXX-XXXX-XXXX,splunkPlatform.index=main,clusterName=spluk-kub" splunk-otel-collector-chart/splunk-otel-collector  

helm install my-splunk-otel-collector --set="splunkPlatform.endpoint=https://prd-p-19lmt.splunkcloud.com:8088/services/collector/event,splunkPlatform.token=XXXX-XXXX-XXXX-XXXX-XXXX,splunkPlatform.index=main,clusterName=spluk-kub" splunk-otel-collector-chart/splunk-otel-collector  

 

Here is how the error log loooks like

 

2022-09-01T04:43:43.953Z info exporterhelper/queued_retry.go:427 Exporting failed. Will retry the request after interval. {"kind": "exporter", "data_type": "logs", "name": "splunk_hec/platform_logs", "error": "Post \"[https://http-inputs-prd-p-19lmt.splunkcloud.com:8088/services/collector/event\](https://http-inputs-prd-p-19lmt.splunkcloud.com:8088/services/collector/event/)": dial tcp: lookup http-inputs-prd-p-19lmt.splunkcloud.com on 10.4.0.10:53: no such host", "interval": "32.444167474s"}
2022-09-01T04:43:52.374Z error exporterhelper/queued_retry.go:176 Exporting failed. No more retries left. Dropping data. {"kind": "exporter", "data_type": "logs", "name": "splunk_hec/platform_logs", "error": "max elapsed time expired Post \"[https://http-inputs-prd-p-19lmt.splunkcloud.com:8088/services/collector/event\](https://http-inputs-prd-p-19lmt.splunkcloud.com:8088/services/collector/event/)": dial tcp: lookup http-inputs-prd-p-19lmt.splunkcloud.com on 10.4.0.10:53: no such host", "dropped_items": 1}
go.opentelemetry.io/collector/exporter/exporterhelper.(*queuedRetrySender).onTemporaryFailure

 

Please advice . Thank you.

**Comment by atoulme** [comment, marked as answer]
This error indicates that your collector cannot connect to the `https://http-inputs-prd-p-19lmt.splunkcloud.com:8088/services/collector/event` endpoint.

Please ensure this endpoint is valid and responds. Currently this host is not registered under a DNS record:
```
$> host http-inputs-prd-p-19lmt.splunkcloud.com                              
Host http-inputs-prd-p-19lmt.splunkcloud.com not found: 3(NXDOMAIN)
```
Please refer to this documentation to establish the correct endpoint associated with your Splunk Cloud instance: https://docs.splunk.com/Documentation/Splunk/9.0.1/Data/UsetheHTTPEventCollector#Send_data_to_HTTP_Event_Collector_on_Splunk_Cloud_Platform

**Comment by oarusoft** [comment]
I did followed that articate before and built my endpoint.

I am using Splunk Cloud trail and my Splunk cloud website is https://prd-p-19lmt.splunkcloud.com/

Based on the docment the sysntax is <protocol>://http-inputs-<host>.splunkcloud.com:<port>/<endpoint>

I modified a bit to use **raw** instead of **event** and still getting error "**no such host**"
Below is what I used.

 helm install my-splunk-otel-collector --set="splunkPlatform.endpoint=https://http-inputs-prd-p-19lmt.splunkcloud.com:8088/services/collector/raw,splunkPlatform.token=xxxxxxxxx,splunkPlatform.index=main,clusterName=splunk" splunk-otel-collector-chart/splunk-otel-collector
NAME: my-splunk-otel-collector

Please suggest what should I use when tail website is https://prd-p-19lmt.splunkcloud.com/

Thanks

**Reply by atoulme** [reply]
Understood. It appears to me that you are correctly using the substitution, yet the subdomain to ingest with HEC is missing. Please consider opening a support case with Splunk to resolve this issue, feel free to point them at this discussion.

**Reply by oarusoft** [reply]
Thank You, Will do that.

**Reply by atoulme** [reply]
After a bit more digging, it appears that your HEC ingest URL is different. I guess you might be using a trial instance.

Please try with:
> https://prd-p-19lmt.splunkcloud.com:8088/services/collector/event

**Reply by oarusoft** [reply]
yeah I am using trail instance.

Here the modifed helm i am using

helm install my-splunk-otel-collector --set="splunkPlatform.endpoint=https://prd-p-19lmt.splunkcloud.com:8088/services/collector/event,splunkPlatform.token=xxxx,splunkPlatform.index=main,clusterName=splunk" splunk-otel-collector-chart/splunk-otel-collector

Now i am getting below error

2022-09-02T04:10:54.950Z        info    service/collector.go:128        Everything is ready. Begin running and processing data.
2022-09-02T04:11:03.846Z        info    exporterhelper/queued_retry.go:427      Exporting failed. Will retry the request after interval.        {"kind": "exporter", "data_type": "logs", "name": "splunk_hec/platform_logs", "error": "Post \"https://prd-p-19lmt.splunkcloud.com:8088/services/collector/event\": x509: certificate is not valid for any names, but wanted to match prd-p-19lmt.splunkcloud.com", "interval": "5.369411187s"}

Please advice

Thank You

**Reply by atoulme** [reply]
Your HEC endpoint is now valid, but the SSL certificate exposed by port 8088 is a self-signed certificate. I think that is a limitation of Splunk Cloud trials. I am looking into a workaround.

**Reply by atoulme** [reply]
Please try `splunkPlatform.insecureSkipVerify=true` to skip TLS certificate verification. Of course, this should not be used in a production setting.

Example:
```helm install my-splunk-otel-collector --set="splunkPlatform.endpoint=https://prd-p-19lmt.splunkcloud.com:8088/services/collector/event,splunkPlatform.token=xxxx,splunkPlatform.index=main,clusterName=splunk,splunkPlatform.insecureSkipVerify=true" splunk-otel-collector-chart/splunk-otel-collector```

**Reply by oarusoft** [reply]
WoW, its working now :) 

Now I did all this because we were getting similar error in production. At this time unfortunetly I don't have access to Prod instance to reproduce this. in about 10 hours I will update exact error we were getting in product.

Thank You @atoulme  for quick help. Really appreciate it.

Regards.

**Reply by oarusoft** [reply]
Quick question, if i am getting the same certificate error in prod instance too then what should be my solution ?

Please advice.
Thank You.

**Reply by atoulme** [reply]
You can perform the same modification and see if it fixes it. It will help make sure the certificate is the problem.

If you choose to ignore the certificate, it means your collector will no longer check the identity of the remote server it connects to.
An attacker could place itself in the middle and intercept data.

**Reply by oarusoft** [reply]
Yeah I agree. Thank you.

**Reply by oarusoft** [reply]
Sorry few more questions.

So what if its indeed certficate problem in  prod and we don't want to use "splunkPlatform.insecureSkipVerify=true".  How do we fix this.

Where can I see other possible splunkPlatform.XXXXX flags . Is there any refrence documets to see all flags

Also I  skipped splunkPlatform.metricsIndex for my test environment. In prod we do have token for metricIndex too.
Below is a helm example with metricsIndex. My question is what is the use of **metricsIndex** token if I am able to use mertic index name directly. Is it so that I can use metrics token directly ?

`helm install my-splunk-otel-collector --set="splunkPlatform.endpoint=127.0.0.1:8088,splunkPlatform.token=xxxxxx,splunkPlatform.**metricsIndex**=k8s-metrics,splunkPlatform.index=main,clusterName=my-cluster" splunk-otel-collector-chart/splunk-otel-collector`

 Is there any splunkbase app that can use  OpenTelemetry  data in splunk to create dashbaord similar to datadog? I don't want to use Splunk Observability at this time.

Thank You.

**Reply by atoulme** [reply]
If the certificate is invalid, please open a support case.

Would you mind separating these other questions into separate issues? This way my team mates can help you.

**Reply by oarusoft** [reply]
In prod I am seeing "HTTP 404 error  \"not found"\". We followed the same guidelines you suggested.  We did added "splunkPlatform.insecureSkipVerify=true" and still got the same error.

Also I mofied the helm ( example below) to inlude  http-inputs and now i am not getting any error. But not seeing any log injested into splunk

`helm install my-splunk-otel-collector --set="splunkPlatform.endpoint=https://http-inputs-<Prod-Endpoint>:8088/services/collector/event,splunkPlatform.token=xxxx,splunkPlatform.index=main,clusterName=splunk,splunkPlatform.insecureSkipVerify=true" splunk-otel-collector-chart/splunk-otel-collector`

Please advice.

**Reply by oarusoft** [reply]
This issue is resolve. I was not sending the channel token.

**Reply by cgreengr** [reply]
Hi @atoulme, I am not using the OTel Collector, but something very similar in Edge Delta's Agent for a free trial and followed the directions of this post as well as disabled_verify: true for the TLS. And am still receiving these errors 

   error.count": 35,
  "error.count.ten_minutes": 35,
  "integration_name": "splunk-test-1",
  "last_error": "Post \"https://prd-p-9vxw3.splunkcloud.com:8088/services/collector/event\": dial tcp: lookup prd-p-9vxw3.splunkcloud.com on 10.4.0.10:53: no such host",
  "last_message": "Incorrect Splunk HEC URL",

Any thoughts?

**Reply by atoulme** [reply]
It looks like your host doesn't exist:
> $> host prd-p-9vxw3.splunkcloud.com
Host prd-p-9vxw3.splunkcloud.com not found: 3(NXDOMAIN)

Are you sure this host name is correct?

**Reply by cgreengr** [reply]
I think I found the error. Thank you so much!

**Reply by cgreengr** [reply]
@atoulme

**Reply by vinitmasaun** [reply]
@atoulme 

We are experiencing a similar error but it doesn't say no such host and we verified that the Otel collector pod is able to communicate with the splunkhec endpoint.  Following is the error printed in the Otel-collector container:

2023-05-02T20:47:19.458Z    error    exporterhelper/queued_retry.go:175    Exporting failed. No more retries left. Dropping data.    {"kind": "exporter", "data_type": "logs", "name": "splunk_hec/platform_logs", "error": "max elapsed time expired Post \"https://<removed_from_public_view>:8088/services/collector\": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)", "dropped_items": 133}

We verified the hec token is correct and are able to use curl with the same token to send test event to the hec endpoint from the collector pod. No logs are being indexed from the k8s cluster due to the above error. We have changed the log level to debug but we couldn't find any additional clue in the debug log statements.  What else could be the cause of this issue?

**Reply by atoulme** [reply]
I am asking our support team to reach out and work with you to examine what is happening. They will ask you questions, if they don't have the information already, around which version of the helm chart and collector you are running, what you are connecting to, etc. I can recommend that you try out v0.76.0 with us if you like as it has performance fixes that can help. We would want to debug the Splunk endpoint as well, especially if you go direct to ingest or go through a HF. Let's continue this discussion directly through support.

**Comment by karanbp24** [comment]
Getting same error: 
2024-08-15T15:38:08.975Z        info    exporterhelper/retry_sender.go:118      Exporting failed. Will retry the request after interval.        {"kind": "exporter", "data_type": "logs", "name": "splunk_hec/platform_logs", "error": "Post \"https://http-inputs-prd-p-1ypci.splunkcloud.com:8088/services/collector/raw\": dial tcp: lookup http-inputs-prd-p-1ypci.splunkcloud.com on 172.20.0.10:53: no such host", "interval": "6.110259216s"}
as per documentation its supposed to have http-inputs before host name. 
Not sure if the documentation has misinformation or I am following is wrong

**Reply by ihari03** [reply]
I am using Splunk Enterprise not cloud.
